Assets/_GameJamCat/Scripts/CatGenerator.cs

```diff
@@ -0,0 +1,159 @@
+using System.Collections.Generic;
+using UnityEngine;
+using Sirenix.OdinInspector;
+using UnityEngine.SceneManagement;
+
+namespace GameJamCat
+{
+
+    /// <summary>
+    /// Scriptable Object for 
+    /// </summary>
+    [CreateAssetMenu]
+    public class CatGenerator : ScriptableObject, ICatFactory
+    {
+
+        #region SHADER_PROPERTY_IDS
+        private readonly int FurMapID = Shader.PropertyToID("_BaseMap");
+        private readonly int FurColorID = Shader.PropertyToID("_MainColor");
+        private readonly int EyeColorID = Shader.PropertyToID("_EyeColor");
+        private readonly int EarColorID = Shader.PropertyToID("_EarColor");
+        private readonly int FeetColorID = Shader.PropertyToID("_FeetColor");
+        private readonly int FurOffsetID = Shader.PropertyToID("_FurOffset");
+        private readonly int FurScaleID = Shader.PropertyToID("_FurScale");
+        private readonly int NoiseWeightID = Shader.PropertyToID("_NoiseWeight");
+        private readonly int NoiseScaleID = Shader.PropertyToID("_NoiseScale");
+        #endregion
+
+
+        [SerializeField]
+        CatBehaviour _catPrefab;
+        [SerializeField]
+        Texture2D[] _furs;
+        [SerializeField, ColorUsage(false)]
+        Color[] _furColors, _feetColors, _earColors;
+        [SerializeField, ColorUsage(false, true)]
+        Color[] _eyeColors;
+
+        [SerializeField, Range(0f, 1f), Tooltip("Percent of cats with textures, relative to solid colors")]
+        float _percentTextured = 0.5f;
+        [SerializeField, MinMaxSlider(-10.0f, 10.0f)]
+        Vector2 _textureOffsetRange = new Vector2(-5f, 5f);
+        [SerializeField, MinMaxSlider(0.0f, 5.0f)]
+        Vector2 _textureScaleRange = new Vector2(1f, 5f);
+        [SerializeField, MinMaxSlider(0.0f, 1.0f)]
+        Vector2 _noiseWeightRange = new Vector2(0f, 1f);
+        [SerializeField, MinMaxSlider(0.0f, 10f)]
+        Vector2 _noiseScaleRange = new Vector2(0f, 5f);
+
+        [SerializeField]
+        bool _poolObjects = false;
+        [System.NonSerialized]
+        List<CatBehaviour> _pool = null;
+        Scene _poolScene;
+
+        public CatBehaviour GetRandomCat()
+        {
+            var newCat =  _poolObjects ? GetRandomCat_pooled() : GetRandomCat_unpooled();
+            SetupCatMaterial(newCat.CatRenderer.material);
+            
+            return newCat;
+        }
+
+        public void Reclaim(CatBehaviour cat)
```

Comment:
Can you rename this method, it's a bit vague as to what it does. Does it remove the cat?  Does it Create and Object Pool?